src/PointwiseFunctions/AnalyticData/GrMhd/MagneticFieldLoop.hpp

```diff
@@ -0,0 +1,218 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <limits>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Minkowski.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/IdealFluid.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/MakeArray.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+// IWYU pragma:  no_include <pup.h>
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace grmhd {
+namespace AnalyticData {
+
+/*!
+ * \brief Analytic initial data for an advecting magnetic field loop.
+ *
+ * This test, originally proposed in \cite Gardiner2005hy and presented in a
+ * slightly modified form by \cite Beckwith2011iy, a region with a circular
+ * cross section with radius `MagneticLoopRadius` is given a non-zero azimuthal
+ * magnetic field of constant magnitude `MagFieldStrength` with zero magnetic
+ * field outside the loop.  This is embedded in an ideal fluid with the given
+ * `AdiabaticIndex`, `RestMassDensity` and `Pressure` with a uniform
+ * `AdvectionVelocity`.  The magnetic field loop should advect across the grid,
```

Comment:
Does the advecting loop test test anything in particular that can be mentioned in this section? Something that can fall under "A failure of the loop advection test (when other grmhd tests pass) might indicate the following: ..."

Reply:
This test is relevant because it diagnoses how well our scheme preserves the no-monopole condition. Since we currently use a divergence cleaning scheme (as opposed as other schemes that preserve div B = 0 to (ideally) machine precision), the expectation is to keep the monopoles small but probably never too small. 

in addition, since it is expected that our code will diffuse some magnetic energy over time, you can consider this test as a demonstration of our scheme's diffusivity for a given resolution.